### ISSUE TITLE
stdClass member warnings

### ISSUE THREAD

**Issue by trante** [themattharris/tmhOAuth#47]
I'm sorry this issue is not directly with tmhOAuth but maybe you can help me.
I use **_phpStorm**_ as IDE. It gives warnings for **_stdClass**_ responses.

For example

```
    $tmhOAuth = new tmhOAuth(array(
        'consumer_key'    => YOUR_CONSUMER_KEY_SAM,
        'consumer_secret' => YOUR_CONSUMER_SECRET_SAM,
        'user_token'      => $atoken1,
        'user_secret'     => $atoken2,
    ));

    $code=$tmhOAuth->request('GET', $tmhOAuth->url('1/followers/ids') );

    if ($code == 200) {
        $responseData = json_decode($tmhOAuth->response['response'],true);
}
```

IDE gives warning for **$tmhOAuth->response['response']**
The warning is **Field 'response' not found in class tmhOAuth**

It supposes that response is a function.
Should i add some lines to declare it as stdClass, if so how can i do without breaking the functionality ?
I have so many requestes in my code, so warnings became huge..
Thank you in advance.

**Comment by themattharris** [comment]
try adding this line to the top of the tmhOAuth class.
`var $response = array();`

so tmhOAuth.php will start something like:

```
class tmhOAuth {
  const VERSION = 0.621;

  var $response = array();
  /**
   * Creates a new tmhOAuth object
   *
   * @param string $config, the configuration to use for this request
   */
  function __construct($config) {

```

if that solves the issue let me know.

**Comment by trante** [comment]
Thank you. Warnings for **response** were disappeared.

But still IDE complains about **request**.
_$code=$tmhOAuth->request('GET', $tmhOAuth->url('1/account/totals') );
Warning: 'void' method 'request' result used_

I suppose this warning is IDE's bug?

**Comment by trante** [comment]
"request" function's warnings are solved when i changed phpDoc of function.
Return tag, from void to int.
Is "void" word typo there ?
Thank you

  /**
- Makes a curl request. Takes no parameters as all should have been prepared
- by the request method
  *
- @return **int** response data is stored in the class variable 'response'
  */

**Comment by themattharris** [comment]
interesting, to be honest i should review those phpdocs. i don't use an IDE which reads them so these ommissions are generally obscured from me.
thanks for sharing your findings.

**Comment by trante** [comment]
Thank you for help.

**Comment by trante** [comment]
Hello, is there a change about this:
https://github.com/themattharris/tmhOAuth/issues/47#issuecomment-5534380

tmhouath->request has no "return" parameter in phpDoc.
Also in document it is written that function "Make an HTTP request using this library. This method doesn't return anything."
But we can get integer status code with `$code=$tmhOAuth->request(...)`
Which gives warning inside IDE.

All errors are suppressed when I add:
`* @return int`
to the phpdoc of request function.
But I think it should be added to commited version?
Or do I miss something?

**Comment by trante** [comment]
My IDE says that both of these two methods are defined as "returns void" inside PHPdoc. But when I use them I get $code as output. So my IDE gives warning. Everytime I download a newer tmhoauth version, I add `@return int` to the top of each function.
`public function request($method, $url, $params=array(), $useauth=true, $multipart=false, $headers=array())
private function curlit()`

**Comment by themattharris** [comment]
thanks. the PHPDoc got left behind when extra features were added. fixing that up now

**Comment by themattharris** [comment]
fixed in 0.7.3 and shipped
